src/backend-api/Controllers/UpdateController.cs

```diff
@@ -229,6 +227,196 @@ public IActionResult EditProgram([FromBody] PostProgramInputModel input, [FromRo
+                                    server.IsAssigned = true;
+                                    UpdateHardwareHistory(true, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                            }
+                        }
+                    }
+                    // checking if any hardware is to be unassigned
+                    if (input.HardwareUnassigned != null)
+                    {
+                        foreach (var hardware in input.HardwareUnassigned)
+                        {
+                            switch (hardware.Type.ToLower())
+                            {
+                                case "monitor":
+                                    var mon = _context.Monitor.Find(hardware.ID);
+                                    mon.EmployeeId = null;
+                                    mon.IsAssigned = false;
+                                    UpdateHardwareHistory(false, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "peripheral":
+                                    var periph = _context.Peripheral.Find(hardware.ID);
+                                    periph.EmployeeId = null;
+                                    periph.IsAssigned = false;
+                                    UpdateHardwareHistory(false, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "computer":
+                                    var comp = _context.Computer.Find(hardware.ID);
+                                    comp.EmployeeId = null;
+                                    comp.IsAssigned = false;
+                                    UpdateHardwareHistory(false, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "server":
+                                    var server = _context.Server.Find(hardware.ID);
+                                    server.EmployeeId = null;
+                                    server.IsAssigned = false;
+                                    UpdateHardwareHistory(false, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                            }
+                        }
+                    }
+
+                    // list to hold the histories of programs that will be added
+                    List<ProgramHistory> programHistories = new List<ProgramHistory>();
+
+                    // checking if any programs are to be assigned
+                    if (input.ProgramAssigned != null)
```

Comment:
Again, is the program assigned going to make new entries for programs currently assigned as well? Or is it the front-end's job to determine which assigned programs are new and which one's were assigned at an earlier time?

Reply:
Front-ends job